addon/services/toast.ts

```diff
@@ -0,0 +1,185 @@
+ * @param {boolean} tapToDismiss - Whether or not a tap on the toast should close it.
+ * @param {Function} onclick - A function to be called when the toast is clicked.
+ */
+export type ToastOptions = {
+  closeButton?: boolean;
+  tapToDismiss?: boolean;
+  onclick?: Function;
+  timeout?: number | 'none';
+};
+
+type ToastFunction = (message: string, title: string, opts?: ToastOptions) => void;
+
+
+const DEFAULT_TOAST_TIMEOUT = 5000;
+
+const DEFAULT_OPTIONS: ToastOptions = Object.freeze({
+  closeButton: true,
+  tapToDismiss: true,
+  onclick: undefined,
+  timeout: undefined
+});
+
+
+export default class Toast extends Service {
+  private _toasts: Element[] = [];
+
+  /**
+   * Display a success toast
+   *
+   * @param {string} message - The Toast's message
+   * @param {string} title - The toast's title
+   * @param {Object} opts - Toast options
+   * @param {boolean} [opts.closeButton=true] - Display a button to close the toast manually.
+   */
+  success: ToastFunction = this._delegate(ToastType.SUCCESS);
+
+  /**
+   * Display an error toast
+   *
+   * @param {string} message - The Toast's message
+   * @param {string} title - The toast's title
+   * @param {Object} opts - Toast options
+   * @param {boolean} [opts.closeButton=true] - Display a button to close the toast manually.
+   */
+  error: ToastFunction = this._delegate(ToastType.ERROR);
+
+  /**
+   * Display a warning toast
+   *
+   * @param {string} message - The Toast's message
+   * @param {string} title - The toast's title
+   * @param {Object} opts - Toast options
+   * @param {boolean} [opts.closeButton=true] - Display a button to close the toast manually.
+   */
+  warning: ToastFunction = this._delegate(ToastType.WARNING);
+
+  /**
+   * Display an information toast
+   *
+   * @param {string} message - The Toast's message
+   * @param {string} title - The toast's title
+   * @param {Object} opts - Toast options
+   * @param {boolean} [opts.closeButton=true] - Display a button to close the toast manually.
+   */
+  info: ToastFunction = this._delegate(ToastType.INFO);
+
+  private _delegate(type: ToastType): ToastFunction {
+    return (message: string, title: string, opts: ToastOptions = DEFAULT_OPTIONS): void => {
+      if (isEmpty(message) && isEmpty(title)) return;
+
+      this._buildToast(type, message, title, { ...DEFAULT_OPTIONS, ...opts });
+    };
+  }
+
+  private _buildToast(type: ToastType, message: string, title: string, opts: ToastOptions): void {
+    const container: Element = this._buildContainer();
+
+    const toast: Element = document.createElement('div');
+    toast.classList.add('toast', 'toast--hidden', `toast-${type}`);
+
+    const titleContainer: Element = this._buildToastPart('div', ['toast-title'], title);
+    const messageContainer: Element = this._buildToastPart('div', ['toast-message'], message);
+
+    let toastChildren: Element[] = [titleContainer, messageContainer];
+
+    if (opts.closeButton) {
```

Comment:
I feel this `_buildToast` method is still too long.
this `closeButton` logic can be moved to a dedicated method

Reply:
@Miexil can't split much i think ahah